apps/tlon-web/rube/run-single-test.ts

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env node
+import * as childProcess from 'child_process';
+import * as fs from 'fs';
+import fetch from 'node-fetch';
+import * as path from 'path';
+
+import { Ship } from './index';
+
+// Get the test file from command line arguments
+const testFile = process.argv[2];
+
+if (!testFile) {
+  console.error('Usage: pnpm e2e:test <test-file.spec.ts>');
+  console.error('Example: pnpm e2e:test chat-functionality.spec.ts');
+  process.exit(1);
+}
+
+// Validate that the test file exists
+// Note: __dirname will be rube/dist when compiled, so we need to go up two levels
+const testPath = path.join(__dirname, '../../e2e', testFile);
+if (!fs.existsSync(testPath)) {
+  console.error(`Test file not found: ${testPath}`);
+  process.exit(1);
+}
+
+let rubeProcess: childProcess.ChildProcess | null = null;
+let isShuttingDown = false;
+
+// Handle cleanup on exit
+process.on('SIGINT', cleanup);
+process.on('SIGTERM', cleanup);
+process.on('exit', cleanup);
+
+async function cleanup() {
+  if (isShuttingDown) return;
+  isShuttingDown = true;
+
+  console.log('\n🧹 Cleaning up...');
+
+  if (rubeProcess && !rubeProcess.killed && rubeProcess.pid) {
+    console.log('Killing rube process...');
+    try {
+      // Kill the entire process group to ensure all child processes are terminated
+      process.kill(-rubeProcess.pid, 'SIGTERM');
+    } catch (error) {
+      console.log('Process already terminated');
+    }
+  }
+
+  // Additional cleanup - kill any remaining processes on our test ports
+  const ports = ['3000', '3001', '3002', '35453', '36963', '38473'];
```

Comment:
can we pull these from the config file instead of hard coding?

Reply:
yep, good point